### DISCUSSION TITLE
Use of HEATConv with HeteroGraph

### DISCUSSION THREAD

**Discussion by SimonCrouzet** [pyg-team/pytorch_geometric#6869] [Q&A]
Hello everyone, and thanks for your work on the package,

I am encountering some troubles regarding the implementation of a model using HEATConv layers to perform message passing on an HeteroGraph. The layers are defined as Heterogeneous Edge-enhanced Graph Attentional operators, and should be useful on heterogeneous graph, but the expected input format make it really hard to be used from an heterograph: `format()` takes as argument `forward(x: [Tensor], edge_index: [Union][Tensor, SparseTensor], node_type: [Tensor], edge_type: [Tensor], edge_attr: [Optional][Tensor])` while when using HeteroGraphs x has the format of a dict {node_type => x}, same for edge_index and edge_attr.

Is there any simple way to use HEATConv with such graph? Or am I just missing the point and we should implement GATConv directly followed by the use of `to_hetero(model, metadata, aggregation_type)` as HEATConv is designed to replicate an attentional mechanisms on graphs originally not heterogeneous?

Thank you for your help,
Best,

**Comment by rusty1s** [comment]
You can use `hetero_data.to_homogeneous()` to merge your heterogeneous graph into a single `edge_index` and `x` representation with corresponding `node_type` and `edge_type` vectors.
